yjit/src/core.rs

```diff
@@ -659,6 +666,9 @@ pub struct BlockRef(Rc<RefCell<Block>>);
 /// Reference-counted pointer to a branch that can be borrowed mutably
 pub type BranchRef = Rc<RefCell<Branch>>;
 
+/// Reference-counted pointer to an entry that can be borrowed mutably
+pub type EntryRef = Rc<RefCell<Entry>>;
```

Comment:
@XrXr has another PR in flight to remove `Rc<RefCell<Branch>>` https://github.com/ruby/ruby/pull/7443/files

Should we try to avoid using it here too?

Reply:
I changed `Rc<RefCell<Entry>>` to `NonNull<Entry>` and renamed a few functions like Alan's PR https://github.com/ruby/ruby/pull/7535/commits/c0b7ad110bf302a22437e22172a680fa8d8d1a44.